### PR TITLE
Add optional parameters to some codecs, fix aborts in clickhouse-compressor with some codecs.

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add optional arguments to codecs `DoubleDelta(bytes_size)`, `Gorilla(bytes_size)`, `FPC(level, float_size)`, it will allow using this codecs without column type in `clickhouse-compressor`.
Fix possible abrots and arithmetic errors in `clickhouse-compressor` with these codecs. Fixes: https://github.com/ClickHouse/ClickHouse/discussions/47262

### CODE REVIEW

**Comment by emirkmo** [comment]
Thank you very much for the quick turnaround. We were able to grab the full builds from CI. Tested the new parameters, and without parameters, for all codecs and several file formats (native, csv, etc.) and data types (int, str, etc.) with and without null including pipeline codecs (e.g., `doubledelta, ZSTD(8)`).

File: docs/en/sql-reference/statements/create/table.md [hunk 1]
```diff
@@ -394,15 +394,15 @@ These codecs are designed to make compression more effective by using specific f
...
+`DoubleDelta(bytes_size)` — Calculates delta of deltas and writes it in compact binary form. Possible `bytes_size` values: 1, 2, 4, 8, the default value for is `sizeof(type)` if equal to 1, 2, 4, or 8. In all other cases, it’s 1. Optimal compression rates are achieved for monotonic sequences with a constant stride, such as time series data. Can be used with any fixed-width type. Implements the algorithm used in Gorilla TSDB, extending it to support 64-bit types. Uses 1 extra bit for 32-byte deltas: 5-bit prefixes instead of 4-bit prefixes. For additional information, see Compressing Time Stamps in [Gorilla: A Fast, Scalable, In-Memory Time Series Database](http://www.vldb.org/pvldb/vol8/p1816-teller.pdf).
```
Comment (al13n321) [inline docs/en/sql-reference/statements/create/table.md:397]: s/for is/is/

Comment (al13n321) [inline docs/en/sql-reference/statements/create/table.md:397]: Not related to this PR, but: s/32-byte/32-bit/ , right?

Same error in `src/Compresion/CompressionCodecDoubleDelta.cpp` and `docs/zh/sql-reference/statements/create.md`

File: src/Compression/CompressionCodecT64.cpp [hunk 2]
```diff
@@ -33,7 +33,7 @@ class CompressionCodecT64 : public ICompressionCodec
         Bit
     };
 
-    CompressionCodecT64(TypeIndex type_idx_, Variant variant_);
+    CompressionCodecT64(std::optional<TypeIndex> type_idx_, Variant variant_);
```
Comment (al13n321) [inline src/Compression/CompressionCodecT64.cpp:36]: Maybe add something like `// type_idx_ is required for compression, but not for decompression.` - took me a bit to figure it out.

File: src/Compression/CompressionCodecT64.cpp [hunk 3]
```diff
@@ -712,7 +715,7 @@ CompressionCodecT64::CompressionCodecT64(TypeIndex type_idx_, Variant variant_)
 void CompressionCodecT64::updateHash(SipHash & hash) const
 {
     getCodecDesc()->updateTreeHash(hash);
-    hash.update(type_idx);
+    hash.update(type_idx ? *type_idx : TypeIndex::Nothing);
```
Comment (al13n321) [inline src/Compression/CompressionCodecT64.cpp:718]: nit: `type_idx.value_or(TypeIndex::Nothing)` is shorter.

**Review by al13n321 — CHANGES_REQUESTED** [review]

File: docs/en/sql-reference/statements/create/table.md (hunk 1 above)
Comment (Avogar) [reply docs/en/sql-reference/statements/create/table.md]: Fixed

Comment (Avogar) [reply docs/en/sql-reference/statements/create/table.md]: Yep, you are right, changed

File: src/Compression/CompressionCodecT64.cpp (hunk 3 above)
Comment (Avogar) [reply src/Compression/CompressionCodecT64.cpp]: Agree, changed

File: src/Compression/CompressionCodecT64.cpp (hunk 2 above)
Comment (Avogar) [reply src/Compression/CompressionCodecT64.cpp:37]: Sure, it will be better. Added

**Review by al13n321 — APPROVED** [review]

**Pull request merged**